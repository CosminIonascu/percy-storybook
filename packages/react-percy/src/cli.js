import * as args from './args';
import pkgDir from 'pkg-dir';
import readPercyConfig from 'react-percy-config';
import { resolve as readWebpackConfig } from 'react-percy-webpack';
import runPercy from './runPercy';
import yargs from 'yargs';

const VERSION = require('../package.json').version;

// eslint-disable-next-line import/prefer-default-export
export function run(argv) {
    argv = yargs(argv)
        .usage(args.usage)
        .help()
        .alias('help', 'h')
        .options(args.options)
        .epilogue(args.docs)
        .argv;

    if (argv.help) {
        yargs.showHelp();
        process.on('exit', () => process.exit(1));
        return;
    }

    if (argv.version) {
        process.stdout.write(`v${VERSION}\n`);
        process.on('exit', () => process.exit(0));
        return;
    }

    const packageRoot = pkgDir.sync();

    const percyConfig = readPercyConfig(packageRoot);
    const webpackConfig = readWebpackConfig(argv.config);

    return runPercy(percyConfig, webpackConfig)
        .then(() => process.on('exit', () => process.exit(0)))
        .catch(() => process.on('exit', () => process.exit(1)));
}
