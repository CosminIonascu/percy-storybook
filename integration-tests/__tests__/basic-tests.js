// eslint-disable-next-line import/no-extraneous-dependencies
import * as percy from 'percy-client';
import { run } from 'react-percy/lib/cli';

jest.mock('percy-client');

process.exit = jest.fn();

it('uploads assets', async () => {
    await run(['--config', 'webpack/webpack.config.js']);

    expect(percy.uploadResource.mock.calls).toMatchSnapshot();
});
