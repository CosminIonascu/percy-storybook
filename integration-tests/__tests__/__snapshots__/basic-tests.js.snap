// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uploads assets 1`] = `Array []`;
